site/app/controllers/student/SubmissionController.php

```diff
@@ -1020,6 +1020,75 @@ public function ajaxRegrade($gradeable_id): array {
         return $this->core->getOutput()->renderResultMessage($msg, true);
     }
 
+    /**
+     * @return JsonResponse
+     * @param string $gradeable_id
+     */
+    #[Route('/api/{_semester}/{_course}/gradeable/{gradeable_id}/values', methods: ['GET'])]
+    public function ajaxGetGradeableValues(string $gradeable_id): JsonResponse {
+        if ($this->core->getQueries()->getUserById($_GET['user_id']) === null) {
```

Comment:
Doesn't this potentially reveal the students in the course?  Shouldn't you instead check to see if the requested user is the current authenticated user first, or otherwise a higher level role which can see the enrollment list?

Reply:
Yes, I didn't consider that, so now the order of the checks has been changed, to account for this.